### ISSUE TITLE
prefix_is_current_class: false positive

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#3755]
Case where class-name is required but abaplint shows error. Often found in constructors.

```
/zfoo.prog.abap[13, 5]: Reference to current class can be omitted: "ZCL_TESTS=>"(prefix_is_current_class)
```

Example:

```abap
CLASS zcl_tests DEFINITION PUBLIC.
  PUBLIC SECTION.
    "! Parameter with same name as class-data
    CLASS-METHODS init
      IMPORTING
        test TYPE i.
  PRIVATE SECTION.
    CLASS-DATA test TYPE i.
ENDCLASS.

CLASS zcl_tests IMPLEMENTATION.
  METHOD init.
    zcl_tests=>test = test.  "<<< can't remove class here
  ENDMETHOD.
ENDCLASS.
```

Similar to #3707 and #1952.
